docs/memory-safety-continuum/memory-safety-continuum-draft2.md

```diff
@@ -0,0 +1,137 @@
+# The Memory Safety Continuum
+
+## Introduction
+
+Much of the conversation around software memory safety has treated the concept as if it were a binary where software is either memory safe or it is not. The truth, as is often the case, is more complex. In an ideal world, we would all write our software in memory safe by default languages, never needing to use an unsafe block or to interface with software written in a non-memory safe by default language. We do not live in that world, however, and are unlikely to live in that world any time soon. Memory safety in software should be considered a continuum, rather than being binary. Within a continuum, even if we are never at the ideal, we can strive to improve with every line of code we write - whether it is net new code, changes to code we are responsible for maintaining or (when appropriate) targeted rewrites of existing code.
+
+In the document, we present the idea of the memory safety continuum - defining various stages and helping you understand where on the continuum the software you write falls today. It is important to understand that different stages may apply to different areas of the same code base. Even if you write the majority of your software in a language like Rust, you will likely still need to interface with existing code written in C or C++, and these interfaces will require different practices for memory safety than other areas of your code base. The primary audience for this document is developers who want to write safer code. It will also likely provide value to engineering managers, product managers, project managers, and anyone else who wants to better understand memory safety in software.
+
+Become more memory safe is a iterative process, no matter where you start on the continuum. This is especially true when dealing with large existing codebases. Small steps, such as introducing memory safe abstractions over non-memory safe code, can make a big difference over time.
+
+### Definitions
+
+The OpenSSF Memory Safety SIG was created with the mission of understanding and reducing memory safety vulnerabilities in Open Source Software. One of our first tasks was to define memory safety for the purpose of our group's work. We considered definitions from various sources and, ultimately, came to a consensus on **[this definition](https://github.com/ossf/Memory-Safety/blob/main/docs/definitions.md)**. While Wikipedia may not traditionally be considered a good source to cite, the crowd editors of [the Memory safety entry](https://en.wikipedia.org/wiki/Memory_safety) have produced a comprehensive classification list of memory safety errors. Memory safety errors are the class of error that memory safe by default languages seek to eliminate.
+
+We also came to a consensus on using the terms "memory safe by default" for languages such as Rust, Go, and C# and "non-memory safe by default" for languages such as C and C++.
```

Comment:
I agree that in a single-threaded environment, Go is safe by default, but since Go is not data race free by construction, [it's possible to violate memory safety in multithreaded contexts](https://research.swtch.com/gorace). Should this merit a different classification or at least an asterisk?